records_mover/cli/job_config_schema_as_args_parser.py

```diff
@@ -60,6 +60,7 @@ def configure_from_properties(self,
                                   properties: JsonSchema,
                                   required_keys: Iterable[str],
                                   prefix: str = '') -> None:
+        self.required_keys = required_keys
```

Comment:
Really not a huge fan of setting self vars outside of `__init__`, but it does make the rest of the code easier. To move it to `__init__` would require making new overloaded init functions that accept more params, which I'm not sure I want to do

Reply:
that makes sense. Does adding a `setter` method to the class make sense? Then, at least, when assessing the class's code, you'd know what potential attributes the class had?

Reply:
Could you explain a bit more? I'm not really sure what you mean

Reply:
Sorry, please ignore last comment, wrote very quickly.

Basically, I wonder if it just makes sense to break out the piece where you assign the `required_keys` attribute into a separate `set_required_keys` method and then just call that method in the `configure_from_properties` method. Not sure it's worth the trouble.

Reply:
yep!